### PR TITLE
Exception handling, async fixes, tests

### PR DESCRIPTION
**Pull request by alertedsnake**
Added some exception logging for some cases where we are getting them in real use but it's not clear why.

Added `await` and `async` keywords in cases where they should have been, but weren't.

A number of tests were broken - some were testing functions which didn't exist anymore, and none of the async stuff was tested properly. This is now all fixed.

Fixed up the `test` Make target so it actually installs things properly and runs tests.

Also bumped versions of dependencies and Alpine to modern times.

### CODE REVIEW

File: rssalertbot/feed.py [hunk 1]
```diff
@@ -144,8 +145,6 @@ async def _handle_fetch_failure(self, title, description):
             description (str): alert description
         """
 
-        self.log.error(f"{title}: {description}")
```
Comment (mgundel) [inline rssalertbot/feed.py:147]: I would definitely expect this method to still be logging errors, as the method description says.

File: rssalertbot/feed.py [hunk 2]
```diff
@@ -127,11 +127,12 @@ async def _fetch(self, session, timeout=10):
...
 
             except concurrent.futures.CancelledError:
-                self._handle_fetch_failure('Timeout', f"Timeout while fetching feed")
+                self._handle_fetch_failure('Timeout', "Timeout while fetching feed")
 
             except Exception as e:
                 etype = '.'.join((type(e).__module__, type(e).__name__))
-                self._handle_fetch_failure('Exception', f"{etype} fetching feed: {e}")
+                await self._handle_fetch_failure('Exception', f"{etype} fetching feed: {e}")
+                self.log.exception("Error fetching feed")
```
Comment (mgundel) [inline rssalertbot/feed.py]: I think it's fine to log this as `error` via `_handle_fetch_failure` in order to retain the alerting functionality.

**Review by mgundel — CHANGES_REQUESTED** [review]

File: rssalertbot/feed.py (hunk 1 above)
Comment (alertedsnake) [reply rssalertbot/feed.py:147]: Ah, I can make that more clear.

File: rssalertbot/feed.py (hunk 2 above)
Comment (alertedsnake) [reply rssalertbot/feed.py]: I specifically wanted the exception trace itself to help diagnose this, and wanted to do it outside of another async call. Though it now occurs to me I should swap the order of lines 134 and 135 to log first.

Comment (mgundel) [reply rssalertbot/feed.py]: In that case, the two other calls to `_handle_fetch_failure` should be updated to log independently of the method call.

**Review by mgundel — APPROVED** [review]

**Pull request merged**